src/App.js

```diff
@@ -14,11 +16,120 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGood: [' Jam'],
+  }
+
+  clear = () => {
+    this.setState({ selectedGood: [] });
+  }
+
+  toAdd = (good) => {
+    this.setState(prevState => (
+      {
+        selectedGood: [...prevState.selectedGood, ` ${good}`],
+      }
+    ));
+  }
+
+  toRemove = (good) => {
+    this.setState(prevState => (
+      {
+        selectedGood: [...prevState.selectedGood]
```

Comment:
```suggestion
        selectedGood: prevState.selectedGood
```
`filter` не мутирующий метод, незачем делать копию